homeassistant/components/device_tracker/__init__.py

```diff
@@ -226,6 +214,8 @@ def async_see_service(call):
     hass.services.async_register(
         DOMAIN, SERVICE_SEE, async_see_service, descriptions.get(SERVICE_SEE))
 
+    # restore
+    yield from tracker.async_setup_tracked_device()
```

Comment:
This flow seems wrong, since the platforms might "see" a device, thereby adding it and the `added_to_hass` will overwrite the state the DB/restore values.

If you really change this (which I'm not sure why), you will probably have to test if it's not already part of the state machine in added_to_hass